File: exkurs/euclidian_distance/dist.py
def euclidean_distance(p1, p2):
    if len(p1) != len(p2):
        raise ValueError("Incompatible points")
    sq_dist = 0
    for dim in range(len(p1)):
        sq_dist += (p2[dim] - p1[dim]) ** 2

    return sq_dist ** 0.5 # sqrt(sq_dist)

File: exkurs/euclidian_distance/dist.js
function euclidianDistance(p1, p2) {
    if(p1.length !== p2.length) {
        throw new Error("Incompatible point dimensions")
    }

    let sqDist = 0;

    for(let dim = 0; dim<p1.length; dim++) {
        sqDist += (p1[dim] - p2[dim]) ** 2;
    }

    return sqDist ** 0.5; // square root
}